### ISSUE TITLE
Install pkgconfig files by default on Linux/FreeBSD

### ISSUE THREAD

**Issue by akien-mga** [SFML/SFML#1075]
Currently the option to install the pkgconfig files is set to False by default. On Linux environments (and I guess FreeBSD too), it's becoming increasingly common to rely on pkgconfig files to detect libraries, so I think it would make sense to make this option default to True.

In this project I maintain, looking for SFML2 using CMake and `find_package(SFML2)` won't work if users just installed SFML2 with `cmake && make` without specifying that they want to install the pkgconfig files (which I guess most end users are not aware of).

(A related problem might be that your FindSFML.cmake file is installed to a non conventional path, but I'll make another issue for that).

**Comment by eXpl0it3r** [comment]
It was originally set to true, but got changed with [this commit](https://github.com/SFML/SFML/commit/93b2152a08f0efc8488fda0c0abfe72834bf79c0), I'm not fully sure what @LaurentGomila meant with the commit message, but maybe he elaborate.

Also please take a look at the [contribution guidelines](http://www.sfml-dev.org/contribute.php) and start discussions on the forum next time, so we can use this tracker for confirmed issues only.

**Comment by LaurentGomila** [comment]
> I'm not fully sure what @LaurentGomila meant with the commit message

If pkg-config files contain global paths such as /home/laurent/sfml, it means that they are not generic and must be generated directly on the target machine. And anything that is user-specific is disabled by default, so that we don't end up with unusable files when we create releases with default options.

**Comment by eXpl0it3r** [comment]
So we won't change this then.

**Comment by akien-mga** [comment]
I don't see any home path in a pkgconfig file installed using default options as you mention:
```
$ cat /usr/lib64/pkgconfig/sfml-all.pc 
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib64
includedir=${prefix}/include

Name: SFML-all
Description: The Simple and Fast Multimedia Library, all modules.
URL: http://www.sfml-dev.org
Version: 2.3.2
Requires: sfml-system, sfml-window, sfml-graphics, sfml-audio, sfml-network
```

It will only contain a home path if you built SFML with ``-DCMAKE_INSTALL_PATH=/home/akien/my_release_deployment_folder``, but then that's not really "default options" anymore, it's a specific use case to make release tarballs, and I don't find it far fetched to have to turn the installation of pkgconfig files off in such cases.

The typical use case would be an end user on Linux that does:
```
cmake
make -j4
sudo make install
```
In such a use case, you do want the pkgconfig file to get installed. If not, it's pointless to have a pkgconfig file if developers using SFML can't rely on the fact that the people who will try to build their game have their system-installed SFML properly installed.

But it's been settled already, so that's fine by me, I guess I'll just have to embed a FindSFML.cmake module in my projects that does not rely on pkgconfig...

**Comment by eXpl0it3r** [comment]
Having spoken with @TankOs, who usually creates the Linux binaries, it seems that installing by default is the better option after all. We just have to be careful when packaging the binaries.

**Comment by binary1248** [comment]
Is this still valid considering #1335 has been merged?

**Comment by akien-mga** [comment]
It's still valid, though it's nice to have advanced CMake config as #1335 implements, it's still CMake-specific. pkgconfig can be used by most buildsystems and parsed by package managers, so it should be installed for any library to simplify its use in applications.

**Comment by AMDmi3** [comment]
I second this issue. Currently I have to specify `-DSFML_INSTALL_PKGCONFIG_FILES=ON` to install the files, which seem superfluous. There are other problems with pkgconfig files, see #1433

**Comment by eXpl0it3r** [comment]
Fixed in 9712bb1ec8379cc29fd7417f64bc5e6b11cc4728
